sandbox/apparmor/apparmor_test.go

```diff
@@ -378,3 +393,45 @@ func (s *apparmorSuite) TestValidateFreeFromAAREhappy(c *C) {
 		c.Check(apparmor.ValidateNoAppArmorRegexp(s), IsNil, Commentf("%q raised an error but shouldn't", s))
 	}
 }
+
+func (s *apparmorSuite) TestUpdateHomedirsTunableMkdirFail(c *C) {
+	restore := apparmor.MockMkdirAll(func(string, os.FileMode) error {
+		return errors.New("mkdir failure")
+	})
+	defer restore()
+
+	err := apparmor.UpdateHomedirsTunable([]string{"does", "not", "matter"})
+	c.Check(err, ErrorMatches, `cannot create AppArmor tunable directory: mkdir failure`)
+}
+
+func (s *apparmorSuite) TestUpdateHomedirsTunableWriteFail(c *C) {
+	restore := apparmor.MockMkdirAll(func(string, os.FileMode) error {
+		return nil
+	})
+	defer restore()
+
+	restore = apparmor.MockAtomicWrite(func(string, io.Reader, os.FileMode, osutil.AtomicWriteFlags) error {
+		return errors.New("write failure")
+	})
+	defer restore()
+
+	err := apparmor.UpdateHomedirsTunable([]string{"does", "not", "matter"})
+	c.Check(err, ErrorMatches, `write failure`)
+}
+
+func (s *apparmorSuite) TestUpdateHomedirsTunableHappy(c *C) {
+	err := apparmor.UpdateHomedirsTunable([]string{"/home/a", "/dir2"})
+	c.Assert(err, IsNil)
+	configFile := filepath.Join(dirs.GlobalRootDir, "/etc/apparmor.d/tunables/home.d/snapd")
+	fileContents, err := ioutil.ReadFile(configFile)
+	c.Assert(err, IsNil)
+	c.Check(string(fileContents), Equals,
+		`# Generated by snapd -- DO NOT EDIT!`+"\n"+`@{HOMEDIRS}+="/home/a" "/dir2"`)
```

Comment:
(nitpick) I personally prefer to write this as: 
```
		`# Generated by snapd -- DO NOT EDIT!
@{HOMEDIRS}+="/home/a" "/dir2"`)
```
but it's a personal preference, fine this way too :)